executor/union_scan.go

```diff
@@ -220,6 +249,30 @@ func (us *UnionScanExec) getSnapshotRow(ctx context.Context) ([]types.Datum, err
 	return us.snapshotRows[0], nil
 }
 
+// For index reader and index look up reader, update doesn't write index to txn memBuffer when the idx column
+// was unchanged. So the `memIndexReader` and `memIndexLookUpReader` can't read the index from txn memBuffer.
+// This function is used to get the missing row by the handle if the handle was in dirtyTable.addedRows.
+func (us *UnionScanExec) getMissIndexRowsByHandle(handle int64) error {
+	reader := us.children[0]
+	switch reader.(type) {
+	case *TableReaderExecutor:
+		return nil
+	}
+	if _, ok := us.dirty.addedRows[handle]; !ok {
+		return nil
+	}
+	// Don't miss in memBuffer reader.
+	if _, ok := us.memIdxHandles[handle]; ok {
+		return nil
+	}
+	ds, err := us.getMemRow(handle)
```

Comment:
The `ds` can be nil if not all conditions in `UnionScanExec` are pushed down.
This results in the `UnionScanExec` return early before getting all results.
This potential issue is found by @jackysp

Reply:
Great, Thanks. 
It's a bug,  already fixed and add test for this case.